### PR TITLE
Adding deprecation checks for geo_shape parameters

### PR DESCRIPTION
**Pull request by masseyke**
The following properties have been removed from the geo_shape data type in 8.0: "strategy", "tree", "tree_levels", "precision", "distance_error_pct", "points_only". This PR adds a deprecation info API check for indexes and templates that have any of these properties in their mappings. Relates to #42404 and #70850.

### CODE REVIEW

**Review by jbaiera — CHANGES_REQUESTED** [review]
A few questions, but things look good

File: x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/ClusterDeprecationChecks.java [hunk 1]
```diff
@@ -201,4 +207,38 @@ static DeprecationIssue checkClusterRoutingAllocationIncludeRelocationsSetting(f
             DeprecationIssue.Level.WARNING
         );
     }
+
+    @SuppressWarnings("unchecked")
+    static DeprecationIssue checkGeoShapeTemplates(final ClusterState clusterState) {
+        String messageForCluster =
+            StreamSupport.stream(clusterState.getMetadata().getTemplates().spliterator(), false).map((templateCursor) -> {
```
[the remark this answers is not in the document]
Comment (masseyke) [reply x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/ClusterDeprecationChecks.java:214]: Are those not returned by clusterState.getMetadata().getTemplates()? How do I get access to component templates?

[the remark this answers is not in the document]
Comment (masseyke) [reply x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/ClusterDeprecationChecks.java:214]: Oh I see now that there's a separate clusterState.getMetadata().componentTemplates() method. I'll add that here.

File: x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecks.java [hunk 2]
```diff
@@ -317,4 +319,40 @@ static DeprecationIssue storeTypeSettingCheck(IndexMetadata indexMetadata) {
...
+            );
+    }
+
+    protected static String formatDeprecatedGeoShapeParamMessage(String type, Map.Entry<?, ?> entry) {
+        String fieldName = entry.getKey().toString();
+        Map<?, ?> value = (Map<?, ?>) entry.getValue();
+        return LegacyGeoShapeFieldMapper.DEPRECATED_PARAMETERS.stream()
+            .filter(deprecatedParameter -> value.containsKey(deprecatedParameter))
+            .map(deprecatedParameter -> String.format(Locale.ROOT, "%s parameter [%s] in field [%s] is deprecated and will be removed in " +
+                "a future version", type, deprecatedParameter, fieldName))
```
[the remark this answers is not in the document]
Comment (masseyke) [reply x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecks.java]: Done.

File: x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecks.java [hunk 3]
```diff
@@ -317,4 +319,40 @@ static DeprecationIssue storeTypeSettingCheck(IndexMetadata indexMetadata) {
...
+            .collect(Collectors.joining("; "));
```
[the remark this answers is not in the document]
Comment (masseyke) [reply x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecks.java:336]: Done.

File: x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/ClusterDeprecationChecks.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (masseyke) [reply x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/ClusterDeprecationChecks.java]: Done.

**Review by jbaiera — APPROVED** [review]
LGTM! Thanks for making those changes

**Pull request merged**